File: regions/system/library/rust/migration/src/schema.rs
#[derive(Debug, Clone, PartialEq)]
pub struct Schema {
    pub tables: Vec<Table>,
}

#[derive(Debug, Clone, PartialEq)]
pub struct Table {
    pub name: String,
    pub columns: Vec<Column>,
    pub indexes: Vec<Index>,
    pub constraints: Vec<Constraint>,
}

#[derive(Debug, Clone, PartialEq)]
pub struct Column {
    pub name: String,
    pub data_type: String,
    pub nullable: bool,
    pub default: Option<String>,
}

#[derive(Debug, Clone, PartialEq)]
pub struct Index {
    pub name: String,
    pub table: String,
    pub columns: Vec<String>,
    pub unique: bool,
}

#[derive(Debug, Clone, PartialEq)]
pub enum Constraint {
    PrimaryKey {
        columns: Vec<String>,
    },
    ForeignKey {
        columns: Vec<String>,
        ref_table: String,
        ref_columns: Vec<String>,
    },
    Unique {
        columns: Vec<String>,
    },
    Check {
        expression: String,
    },
}

File: regions/system/library/typescript/migration/src/schema.ts
export interface Column {
  name: string;
  dataType: string;
  nullable: boolean;
  default?: string;
}

export interface Index {
  name: string;
  table: string;
  columns: string[];
  unique: boolean;
}

export type Constraint =
  | { type: 'primaryKey'; columns: string[] }
  | { type: 'foreignKey'; columns: string[]; refTable: string; refColumns: string[] }
  | { type: 'unique'; columns: string[] }
  | { type: 'check'; expression: string };

export interface Table {
  name: string;
  columns: Column[];
  indexes: Index[];
  constraints: Constraint[];
}

export interface Schema {
  tables: Table[];
}
